### PR TITLE
Ready: add org.paraview.ParaView

### PR DESCRIPTION
**Pull request by kevinsmia1939**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** No response, but aware https://discourse.paraview.org/t/i-flatpak-paraview-any-thought/6698
https://discourse.paraview.org/t/where-to-put-example-visualization-files/6753
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView.yaml

**Comment by kevinsmia1939** [comment]
Why is bot attempt to build .json? It is .yaml

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView

**Comment by kevinsmia1939** [comment]
I don't understand why org.paraview.ParaView.appdata.xml is not found.

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView

**Comment by kevinsmia1939** [comment]
I don't know why appstream validate is failing, need help on this.

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView

**Comment by kevinsmia1939** [comment]
I don't know what the error mean, any suggestion?

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView

File: org.paraview.ParaView.yaml [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+      - -DVTK_BUILD_QT_DESIGNER_PLUGIN:BOOL=OFF
+      - -DPARAVIEW_INSTALL_DEVELOPMENT_FILES:BOOL=ON
+      - -DPARAVIEW_BUILD_WITH_EXTERNAL:BOOL=OFF
+      - -DPARAVIEW_ENABLE_FFMPEG:BOOL=ON
+      - -DPARAVIEW_ENABLE_VISITBRIDGE:BOOL=ON
+      #- -DPARAVIEW_USE_CUDA:BOOL=ON Not sure how to do this
+      - -DPARAVIEW_USE_MPI:BOOL=ON
+    post-install:
+      - install -Dm644 /app/share/appdata/$FLATPAK_ID.appdata.xml /app/share/metainfo/$FLATPAK_ID.appdata.xml
+      #- install -Dm644 $FLATPAK_ID.appdata.xml /app/share/appdata/$FLATPAK_ID.appdata.xml
```
Comment (hfiguiere) [inline org.paraview.ParaView.yaml:183]: not sure what you are trying to do here, but given that you ship the appdata.xml you have to install it.
Replace lines 182-183 with something like `install -Dm644 -t $FLATPAK_DEST/share/metainfo org.paraview.ParaView.appdata.xml`

(this should fix the failure)

**Comment by hfiguiere** [comment]
is there a reason why this is x86_64 only?

**Comment by kevinsmia1939** [comment]
> is there a reason why this is x86_64 only?

Disable on aarch64 since GLES isn't supported for bundled vtk but is needed for paraview

File: org.paraview.ParaView.yaml (hunk 1 above)
Comment (kevinsmia1939) [reply org.paraview.ParaView.yaml:183]: Paraview ship its own appdata.xml, but does not conform to the standard so I ship my slightly modified appdata. I will try as you suggest.

Comment (hfiguiere) [reply org.paraview.ParaView.yaml:183]: if it ships it then you should patch it instead... and submit the patch upstream.

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView

File: fix_appdata_file.patch [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ <component type="desktop-application">
+  <id>org.paraview.paraview</id>
+  <metadata_license>CC-BY-4.0</metadata_license>
+- <project_license>BSD</project_license>
++ <project_license>BSD-3-Clause</project_license>
+  <name>ParaView</name>
+  <summary>Data analysis and visualization</summary>
++ <content_rating type="oars-1.1"/>
++ <releases>
++     <release version="5.9.0" date="2021-01-28">
```
Comment (hfiguiere) [inline fix_appdata_file.patch:14]: This cause the file to be invalid xml. `<release>` and `<releases>` are not closed. Hence the build failure.

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView

**Comment by kevinsmia1939** [comment]
The build fail because Paraview is missing 128x128 icon ?

**Comment by hfiguiere** [comment]
> The build fail because Paraview is missing 128x128 icon ?

Yes. Or a SVG version. 

There are options if upstream doesn't have any.

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView

**Comment by nedrichards** [comment]
In testing this doesn't seem to use the file portal, I had to use a --filesystem permission to be able to open files.

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView

**Comment by kevinsmia1939** [comment]
The examples does not seems to be working as well

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView

**Comment by kevinsmia1939** [comment]
I don't know why, but example visualization does not work anymore, it won't open.

```
Your installation doesn't have datasets to load the example visualizations. You can manually download the datasets from paraview.org and then place them under the following path for examples to work:

'/examples'
```

**Comment by kevinsmia1939** [comment]
Example visualization used to work, it does not now, don't know what happen.

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView

**Comment by kevinsmia1939** [comment]
bot, build org.paraview.ParaView

File: org.paraview.ParaView.yaml [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+      - -DPARAVIEW_USE_MPI:BOOL=ON
+    sources:
+      - type: archive
+        url: https://www.paraview.org/files/v5.9/ParaView-v5.9.0.tar.gz
+        sha256: f96c9c996903b295dcfb7db689411d1d0e6a3ccaeb6935c70deb8cc9e5e657c6
+      - type: patch
+        path: fix_appdata_file.patch
+        # ParaView is missing 128x128 icon.
+      - type: patch
+        path: add_icon_128x128.patch
```
Comment (barthalion) [inline org.paraview.ParaView.yaml:188]: Ideally you should try to send these patches upstream.

Comment (kevinsmia1939) [reply org.paraview.ParaView.yaml:188]: Yes, I already did.

**Pull request closed without merging**